src/python/pants/option/global_options.py

```diff
@@ -547,6 +549,33 @@ def register_bootstrap_options(cls, register):
             "Pants's own code, plugins, and `--pants-config-files` are inherently invalidated.",
         )
 
+        process_execution_local_parallelism = "--process-execution-local-parallelism"
+        rule_threads_core = "--rule-threads-core"
+        rule_threads_max = "--rule-threads-max"
+
+        register(
```

Comment:
The order of options does matter: it sets the order in the help system. I recommend moving these right next to the local-parallelism options.

Reply:
The ordering was _somewhat_ intentional. The `--process-*` options are rightly clustered together, but these don't deal with process execution.